docs/developers/guides/token.md

```diff
@@ -0,0 +1,1287 @@
+[Contract Exit] Cannot transfer to self
+ [Success]: ✔ should not transfer tokens to self RTrace: +234
+[Event] koinos.contracts.token.mint_event / [ '1DQzuCcTKacbs9GGScRTU1Hc8BsyARTPqG' ] / ChkAiCtotTmdMTyQZ/OEFgfmKMEtVCA8jEc3EHs=
+[Contract Exit] 'from' has insufficient balance
+ [Success]: ✔ should not transfer if insufficient balance RTrace: +307
+
+    [File]: assembly/__tests__/Token.spec.ts
+  [Groups]: 2 pass, 2 total
+  [Result]: ✔ PASS
+[Snapshot]: 0 total, 0 added, 0 removed, 0 different
+ [Summary]: 13 pass,  0 fail, 13 total
+    [Time]: 139.55ms
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  [Result]: ✔ PASS
+   [Files]: 1 total
+  [Groups]: 2 count, 2 pass
+   [Tests]: 13 pass, 0 fail, 13 total
+    [Time]: 5130.905ms
+┌───────────────────┬───────┬───────┬──────┬──────┬───────────┐
+│ File              │ Total │ Block │ Func │ Expr │ Uncovered │
+├───────────────────┼───────┼───────┼──────┼──────┼───────────┤
+│ assembly/Token.ts │ 100%  │ 100%  │ 100% │ 100% │           │
+├───────────────────┼───────┼───────┼──────┼──────┼───────────┤
+│ total             │ 100%  │ 100%  │ 100% │ 100% │           │
+└───────────────────┴───────┴───────┴──────┴──────┴───────────┘
+
+```
+
+---
+## Customizing the token
+Let's customize the specifics of our token project by modifying `assembly/Token.ts`. Define the following:
+
+- `_name`: The name of your token
+
+- `_symbol`: The symbol for your token
+
+- `_decimals`: The decimal places for your token
+
+
+```ts linenums="1" title="assembly/Token.ts"
+- _name: string = "My Token Name";
+- _symbol: string = "MTN";
+- _decimals: u32 = 8;
+```
+
+As an example, we changed the `[token name]` variable to `My Token Name` and the `symbol` variable to `MTN`.
+
+
+After making your changes, update the token name and symbol in the unit test file located at `assembly/__tests__/Token.spec.ts`.
+We will need to change the test for the `name` and `symbol` functions to reflect your token name and symbol.
+
+```ts linenums="25" title="assembly/__tests__/Token.spec.ts"
+  it("should get the name", () => {
+    const tkn = new Token();
+
+    const args = new token.name_arguments();
+    const res = tkn.name(args);
+
+    expect(res.value).toBe("[token name]");
+  });
+```
+On line 31, replace `[token name]` with the same token name that was entered in the `assembly/Token.ts`
+
+```ts linenums="34" title="assembly/__tests__/Token.spec.ts"
+  it("should get the symbol", () => {
+    const tkn = new Token();
+
+    const args = new token.symbol_arguments();
+    const res = tkn.symbol(args);
+
+    expect(res.value).toBe("[token symbol]");
+  });
+```
+On line 40, replace `[token symbol]` with the same symbol that was entered in the `assembly/Token.ts`
+
+---
+## Compiling the contract
+Once you've made all the necessary modifications and your tests are passing, you're ready to build the release version of your contract. Let's run the build script for release.
```

Comment:
Before making changes to the test, it might be worth running the tests once with the failure, fix the tests, and walk through building and testing once more with the tests passing again. Create a mini debug scenario and walk them through it.